testsuite/single/tests/efuns/dump_prog.c

```diff
@@ -1,3 +1,25 @@
+void internal(int flag) {
+    string fn;
+
+    switch (flag) {
+        case 1:
+            fn = "/PROG_DUMP_1";
+            break;
+        case 2:
+            fn = "/PROG_DUMP_2";
+            break;
+        case 3:
+            fn = "/PROG_DUMP_3";
+            break;
+        default:
+            dump_prog(this_object());
+            return;
+    }
+    dump_prog(this_object(), flag, fn);
```

Comment:
why do we need a switch but not the same style as the test for replace_string ? I feel it is much easier to read that way, just repeat a few lines that call dump_program() with different flags and arguments. 

Also, if we can't verify the content in anyway , at least we could verify the file exists and non-empty after calling this efun.